### PR TITLE
Thoughtless fixes for beta 6

### PR DESCRIPTION
**Pull request by Anviking**

### CODE REVIEW

File: Sources/DecodingError.swift [hunk 1]
```diff
@@ -110,9 +110,9 @@ public func ~=<T>(lhs: T.Type, rhs: Any.Type) -> Bool {
 
 // FIXME: I'm not sure about === equality
 public func ==(lhs: DecodingError.Metadata, rhs: DecodingError.Metadata) -> Bool {
-    return lhs.object === rhs.object
+    return lhs.object as AnyObject === rhs.object as AnyObject
```
Comment (NachoSoto) [inline Sources/DecodingError.swift]: I don't think this makes much sense if they're not objects.

File: Tests/DictionaryTests.swift [hunk 2]
```diff
@@ -10,11 +10,11 @@ import XCTest
 import Decodable
 
 private struct AccessibilityInfo: Decodable {
-    let data: [String: AnyObject]
+    let data: [String: Any]
     
-    private static func decode(_ json: AnyObject) throws -> AccessibilityInfo {
```
Comment (NachoSoto) [inline Tests/DictionaryTests.swift]: This should be `fileprivate`.

File: Tests/ErrorPathTests.swift [hunk 3]
```diff
@@ -12,7 +12,7 @@ import XCTest
 private struct Color: Decodable {
     let name: String
     
-    private static func decode(_ json: AnyObject) throws -> Color {
+    static func decode(_ json: Any) throws -> Color {
```
Comment (NachoSoto) [inline Tests/ErrorPathTests.swift]: And these

File: Tests/ErrorPathTests.swift [hunk 4]
```diff
@@ -96,17 +96,17 @@ class ErrorPathTests: XCTestCase {
     
     func testFoo() {
         let dictionary: NSDictionary = ["key": ["test": 3]]
-        let a: Int = try! uppercase(dictionary => "key" as! NSDictionary) as AnyObject => "TEST"
+        let a: Int = try! uppercase(dictionary => "key" as! NSDictionary) as Any => "TEST"
         XCTAssertEqual(a, 3)
     }
     
     private func uppercase(_ json: NSDictionary) -> NSDictionary {
```
Comment (NachoSoto) [inline Tests/ErrorPathTests.swift]: You probably want to change these types to `[String: Any]` now.

File: Sources/DecodingError.swift (hunk 1 above)
Comment (Anviking) [inline Sources/DecodingError.swift]: @NachoSoto In what sense? If they're not objects they get wrapped in different objects and the reference equality fails. Not 100% sure this works as intended with NSDictionaries though, as there is no test coverage of this.

Also, I appreciate your comments – thanks!

File: Tests/DictionaryTests.swift (hunk 2 above)
Comment (Anviking) [inline Tests/DictionaryTests.swift]: Why? The struct is already `private`.

File: Tests/ErrorPathTests.swift (hunk 4 above)
Comment (Anviking) [inline Tests/ErrorPathTests.swift]: In the general case - perhaps (#118), but it turned out both casts could be removed here. (7135fe9e378561d8523725b98af8db5f3c52e60b)

**Pull request merged**